tests/test_config/test_partial_config.py

```diff
@@ -0,0 +1,118 @@
+import json
+import tempfile
+
+import pytest
+from libsonata._libsonata import SonataError
+
+import bluepysnap.circuit as test_module
+from bluepysnap.exceptions import BluepySnapError
+
+from utils import copy_config, edit_config, copy_test_data
+
+
+def test_partial_circuit_config_wrong_status():
+    with copy_config("circuit_config.json") as config_path:
+        with edit_config(config_path) as config:
+            config["metadata"] = {"status": "NOT A TYPE"}
+
+        with pytest.raises(SonataError):
+            test_module.Circuit(config_path)
+
+
+@pytest.mark.parametrize(
+    "config",
+    [
+        {"metadata": {"status": "partial"}},
+        {
+            "metadata": {"status": "partial"},
+            "components": {
+                "morphologies_dir": "some/morph/dir",
+            },
+            "networks": {
+                "nodes": [],
+            },
+        },
+        {
+            "metadata": {"status": "partial"},
+            "components": {
+                "morphologies_dir": "some/morph/dir",
+            },
+            "networks": {
+                "edges": [],
+            },
+        },
+    ],
+)
+def test_partial_circuit_config_empty(config):
+    with tempfile.NamedTemporaryFile(mode="w+") as config_file:
+        config_file.write(json.dumps(config))
+        config_file.flush()
+        circuit = test_module.Circuit(config_file.name)
+
+    with pytest.raises(BluepySnapError):
+        assert circuit.get_node_population_config("default")
+    with pytest.raises(BluepySnapError):
+        assert circuit.get_edge_population_config("default")
+
+    assert circuit.nodes
+    assert circuit.nodes.population_names == []
+    # property_values returns empty sets if values is empty
+    assert list(circuit.nodes.values()) == []
+    with pytest.raises(BluepySnapError):
+        circuit.nodes.ids()
+
+    assert circuit.edges
+    assert circuit.edges
+    assert circuit.edges.population_names == []
+    # property_values returns empty sets if values is empty
+    assert list(circuit.edges.values()) == []
+    with pytest.raises(BluepySnapError):
+        circuit.edges.ids()
+
+
+@pytest.mark.parametrize(
+    "nodes_update, edges_update",
+    [
+        ({"populations": {"default": {}}}, {"populations": {"default": {}}}),
+    ],
+)
+def test_partial_circuit_config_partial(nodes_update, edges_update):
```

Comment:
I did not figure out other ways to make the configuration partial, ideas are welcome

It seems that the `type` is guessed and everything is working.